packages/devtools/lib/src/performance/performance_controller.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../profiler/cpu_profile_model.dart';
+import '../profiler/cpu_profile_service.dart';
+
+class PerformanceController {
+  final CpuProfilerService cpuProfilerService = CpuProfilerService();
+
+  /// Final cpu profile data after merging the recorded profiles.
+  CpuProfileData cpuProfileData;
+
+  /// Cpu profiles recorded on an interval during the time between starting and
+  /// stopping the recording.
+  List<CpuProfileData> recordedProfiles = [];
+
+  Timer timer;
+
+  bool get recording => _recording;
+
+  bool _recording = false;
+
+  void startRecording() async {
+    reset();
+    _recording = true;
+
+    // TODO(kenzie): uncomment this code once getVMTimelineMicros is available.
```

Comment:
capability detect for this method based on the vm service version and default to zero otherwise.
We should also not be polling the VM every 100ms for this. Getting the time at the start of the run and then assuming the clocks are sufficiently in sync seems sufficient./